### DISCUSSION TITLE
Use ADC with threshold instead of digital input

### DISCUSSION THREAD

**Discussion by digitalbirdo** [jensrossbach/esphome-ferraris-meter#18] [Ideas & Requests]
I had a lot of trouble correctly adjusting the potentiometer on the TCRT5000. Even a modification to a Schmitt-Trigger did not help to get consistant results.

I did a quick hack to read the analog output of the TCRT5000 and switch an output pin to on/off.
I connected GPIO4 diretcly to GPIO5.

It would be a bit nicer to have the possibility of an ADC as an input with an configurable threshold instead of fiddling around with the potentiometer :)

```
sensor:
  - platform: adc
    pin: GPIO17
    id: ferraris_adc
    name: "Ferraris_Raw"
    internal: true
    raw: true
    samples: 10
    update_interval: 50ms
    on_value:
     then:
       lambda: |-
         if (id(ferraris_adc).state < id(ferraris_adc_threshold).state) {
             id(gpio_d1).turn_on();
         }
         else {
             id(gpio_d1).turn_off();
         }

number:
  - platform: template
    id: ferraris_adc_threshold
    name: ADC Schwellwert
    icon: mdi:counter
    entity_category: config
    mode: box
    optimistic: true
    restore_value: true
    initial_value: 58
    min_value: 0
    max_value: 1000
    step: 1

switch:
  - platform: gpio
    # connected to GPIO4 which serves as input for the ferraris component
    pin: GPIO5
    id: gpio_d1
```

**Comment by jensrossbach** [comment]
Yes, that sounds like a good idea. I will have some thoughts about it.

**Reply by jensrossbach** [reply]
Idea has been implemented with release [1.3.0](https://github.com/jensrossbach/esphome-ferraris-meter/releases/tag/v1.3.0).
